packages/kbn-management/settings/components/field_input/input/code_editor_input.tsx

```diff
@@ -90,14 +90,17 @@ export const CodeEditorInput = ({
   );
 
   const debouncedUpdateValue = useMemo(() => {
-    // Trigger update 1000 ms after the user stopped typing to reduce validation requests to the server
-    return debounce(updateValue, 1000);
+    // Trigger update 500 ms after the user stopped typing to reduce validation requests to the server
+    return debounce(updateValue, 500);
   }, [updateValue]);
 
   const onChange: CodeEditorProps['onChange'] = async (newValue) => {
-    // @ts-expect-error
-    setValue(newValue);
-    await debouncedUpdateValue(newValue, onUpdate);
+    // Only update the value when the onChange handler is called with a different value from the current one
+    if (newValue !== value) {
+      // @ts-expect-error
```

Comment:
I'm curious why there is a TS error here and if we could fix it?

Reply:
Thanks for the review @yuliacech! I see that there are two TS errors in the code editor component and I think they are caused because this component is used by both json and markdown type settings, while the [`getFieldInputValue`](https://github.com/elastic/kibana/blob/6dd9a88f013dd369c5b7e88ac20d661e8bdbcf18/packages/kbn-management/settings/components/field_input/input/code_editor_input.tsx#L50) function expects a parameter of a typed setting field. I can't think of an easy fix at this moment so I opened https://github.com/elastic/kibana/issues/177998 to track this work.